### DISCUSSION TITLE
[Bug] Error loading a value into a <input type="date" /> and <input type="datetime-local" />

### DISCUSSION THREAD

**Discussion by gaiustemple** [livewire/livewire#8894] [7. Bugs (archived)]
### Livewire version

v3.5.8

### Laravel version

v11.25.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Try setting a date/datetime-local property's initial value using a [supported php date type](https://livewire.laravel.com/docs/properties#common-php-types) or loading a timestamp value from a Model
2. Whilst the property is echo'ed by blade correctly, the date/datetime-local input value is blank
3. Changing the date/datetime from the input updates the livewire property, however the input goes blank again

The [wirebox](https://wirebox.app/b/4ko9d) may explain it better

@joshhanley Not sure if related to #8115 but the [docs](https://livewire.laravel.com/docs/properties#common-php-types) say that we can do the below code - however as the wirebox shows, this isn't working (unless I am being dumb here which is quite possible, or it's UK browser formatting).
```
$this->date = new DateTime('now'); // DateTime 
$this->date = new Carbon('now'); // Carbon
```

https://github.com/user-attachments/assets/0c67f708-150f-4bbb-8bb1-5838ef59c41f

### Repository or Wirebox URL

https://wirebox.app/b/4ko9d

### Additional information

Issue is occurring on Safari, and Chrome v129.0.6668.71
I am based in UK if that makes any difference to the browser date/time formatting

datetime error
`livewire.js?id=923613aa:2006 The specified value "2024-09-27T21:36:58+00:00" does not conform to the required format.  The format is "yyyy-MM-ddThh:mm" followed by optional ":ss" or ":ss.SSS".`

date error
`The specified value "2024-09-19T00:00:00+00:00" does not conform to the required format, "yyyy-MM-dd".
Fl @ livewire.js?id=d38cabc2:6
Xi @ livewire.js?id=d38cabc2:6
(anonymous) @ livewire.js?id=d38cabc2:6
D @ livewire.js?id=d38cabc2:2
e._x_forceModelUpdate @ livewire.js?id=d38cabc2:6
(anonymous) @ livewire.js?id=d38cabc2:6
r @ livewire.js?id=d38cabc2:6
Qa @ livewire.js?id=d38cabc2:2Understand this warning`

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by lrljoe** [comment]
Please name those properties more clearly to help people understand the issue

Flicking back and forth to see what datetime7 vs datetime4 is not going to encourage people to help.

Give the properties names that match what is expected 

datetime-yymmdd
datetime-raw
etc

and people will definitely help more readily.  I know I don't want to spend 30 minutes making notes as to what each property's format is!

**Reply by gaiustemple** [reply]
Apologies, changed and added below comment to expound further.

**Comment by gaiustemple** [comment]
Thanks for the reply @lrljoe .  I have tried to label the properties better on the [linked Wirebox](https://wirebox.app/b/4ko9d).
There is one example I can't put on Wirebox which is possibly the most impactful, and that is when using Eloquent Models and casts, but I have tried to demonstrate it as comments and below.

This highly impacts trying to edit a Model's date or datetime property as the input doesn't show the user the currently set date or datetime.

When dates are passed between the front and back end as strings, the issue doesn't seem to occur; e.g. when I use `format('Y-m-d')`   However as soon as the property php type is either `DateTime` or `Carbon`, the issue occurs.  So I think that the issue is being caused when [CarbonSynth](https://github.com/livewire/livewire/blob/main/src/Mechanisms/HandleComponents/Synthesizers/CarbonSynth.php) is being used.

In addition to the Wirebox, below is two ways you can easily replicate this:

### Simple date, initiated just like the docs
```php
// Livewire component
namespace App\Livewire;
class Simple extends Component
{
    public function mount()
    {
        $this->date = new DateTime('now'); // DateTime
    }
}
```
```blade
// Livewire blade
<input type="date" wire:model.live="date" />
```
Results in a blank date component
![image](https://github.com/user-attachments/assets/204be8a5-ee35-43f4-b32e-65ab888b56e4)



### Eloquent Model property when cast to a php DateTime

```php
// User Model
class User extends Authenticatable
{
    protected function casts(): array
    {
        return [
            'some_date' => 'date',
            'email_verified_at' => 'datetime',
            'password' => 'hashed',
        ];
    }
}

// Livewire component
namespace App\Livewire;
class Simple extends Component
{
    #[Validate('date')]
    public $date;
    #[Validate('date')]
    public $datetime;

    public function mount()
    {
        $this->date = \App\Models\User::first()->some_date; // Cast to DateTime because of Eloquent casts property
        $this->datetime = \App\Models\User::first()->email_verified_at; // Cast to DateTime because of Eloquent casts property
    }
}
```
```blade
// Livewire blade
<input type="date" wire:model.live="date" />
<input type="datetime-local" wire:model.live="datetime" />
```
Also results in blank date and datetime-local component

![image](https://github.com/user-attachments/assets/b876524c-b3cd-4c92-aa1e-6b711520741b)
![image](https://github.com/user-attachments/assets/04c842c6-f876-4ea4-991a-da7214d59332)

Does this make the issue any clearer?

**Reply by EriksonEnterprises** [reply]
I've always had to format it to work with the two way binding and show the value

It's just an extra step but it works

Regular html without Livewire requires it to be formatted as well (try filling the value of a html form and it won't work unless it's formatted)

**Reply by gaiustemple** [reply]
Sure, but then it is a string and no longer a php DateTime which the docs say is a supported type.  Other than date/datetime inputs, what purpose would the synthesiser `CarbonSynth` have?
